hivemind/moe/server/dht_handler.py

```diff
@@ -93,9 +93,9 @@ async def _get_experts(
     num_workers = len(uids) if dht.num_workers is None else min(len(uids), dht.num_workers)
     found: Dict[ExpertUID, DHTValue] = await node.get_many(uids, expiration_time, num_workers=num_workers)
 
-    experts: List[Optional[RemoteExpert]] = [None] * len(uids)
+    experts: List[Optional[RemoteExpertInfo]] = [None] * len(uids)
     for i, uid in enumerate(uids):
-        expert_info_for_uid = found[uid]
-        if expert_info_for_uid is not None and isinstance(expert_info_for_uid.value, tuple):
-            experts[i] = RemoteExpertInfo(uid, PeerInfo.from_tuple(expert_info_for_uid.value))
+        expert_server_peer_id = found[uid]
+        if expert_server_peer_id is not None and isinstance(expert_server_peer_id.value, str):
+            experts[i] = RemoteExpertInfo(uid, PeerID.from_base58(expert_server_peer_id.value))
```

Comment:
nit: Is "expert server peer" == "expert peer"? If it is, let's keep it simple.

```suggestion
        expert_peer_id = found[uid]
        if expert_peer_id is not None and isinstance(expert_peer_id.value, str):
            experts[i] = RemoteExpertInfo(uid, PeerID.from_base58(expert_peer_id.value))
```

Reply:
fixed